lib/elastomer/client/docs.rb

```diff
@@ -265,12 +272,44 @@ def count( query, params = nil )
       #
       # See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/docs-delete-by-query.html
       #
-      # Returns the response body as a hash
+      # Returns a Hash of statistics about the delete operations
       def delete_by_query( query, params = nil )
         query, params = extract_params(query) if params.nil?
 
-        response = client.delete '/{index}{/type}/_query', update_params(params, :body => query, :action => 'docs.delete_by_query')
-        response.body
+        responses = []
+        accumulate = lambda { |response| responses << response unless response == nil }
+
+        accumulate.call(bulk do |b|
+          scan(query, params).each_document do |hit|
+            accumulate.call(b.delete(_id: hit["_id"]))
```

Comment:
This fails for Docs objects without a given type because the `_type` isn't automatically filled in:

```
Elastomer::Client::RequestError: ActionRequestValidationException[Validation Failed: 1: type is missing;2: type is missing;3: type is missing;4: type is missing;]
```

To avoid that, always pass in the `_type`. It's probably also a good idea to pass in the `_index` as well in case it's possible to call delete_by_query without index scope.

```ruby
b.delete(_id: hit["_id"], _type: hit["_type"], _index: hit["_index"])
```

Another wrinkle: Indices with document routing enabled require the `_routing` parameter to delete. You'll probably want to check for that in the scan response and pass it in.